pkg/scheduler/scheduler.go

```diff
@@ -114,3 +123,83 @@ type VPod interface {
 
 	GetResourceVersion() string
 }
+
+type ScaleCache struct {
+	lock                 sync.RWMutex //protects access to entries, entries itself is concurrency safe, so we only need to ensure that we correctly access the pointer
+	entries              *cache.Expiring
+	statefulSetClient    clientappsv1.StatefulSetInterface
+	statefulSetNamespace string
+}
+
+type scaleEntry struct {
+	specReplicas int32
+	statReplicas int32
+}
+
+func NewScaleCache(ctx context.Context, namespace string) *ScaleCache {
```

Comment:
To be future proof I would create a `ScaleCacheConfig` struct and use that as input to this function, so that we could also configure the hardcoded expire time (now at `time.Minute*5`).

In addition, I would get the `ScaleCacheConfig` from the `SchedulerConfig`